Pool/Home.vb

```diff
@@ -29,7 +37,31 @@ Public Class Home
             .User = txtUser.Text.ToString
             .Password = txtPassword.Text.ToString
         End With
+        'Check if inputs are entered
         If Not String.IsNullOrEmpty(userAuthenticate.User) AndAlso Not String.IsNullOrWhiteSpace(userAuthenticate.Password) Then
+            Dim userOptionsDS As New DataSet
+            userOptionsDS = userAuthenticate.SelUserOptions
+            CurrentSession.TwoFactorEnabled = Convert.ToBoolean(Convert.ToInt16(userOptionsDS.Tables(0).Rows(0).Item("twoFactorAuth").ToString))
```

Comment:
We're not supposed to have an empty dataset since we already pass step 1. In the case, we do, logging here while adding a check before the Int16 cast.

Reply:
A check has been added for empty datasets.